src/main/java/hudson/plugins/scm_sync_configuration/strategies/model/PatternsEntityMatcher.java

```diff
@@ -22,12 +27,33 @@ public boolean matches(Saveable saveable, File file) {
 		if (file == null) {
 			return false;
 		}
-		String filePathRelativeToHudsonRoot = JenkinsFilesHelper.buildPathRelativeToHudsonRoot(file);
-        AntPathMatcher matcher = new AntPathMatcher();
-        for(String pattern : includesPatterns) {
-            if(matcher.match(pattern, filePathRelativeToHudsonRoot)){
-                return true;
-            }
+		return matches(saveable, JenkinsFilesHelper.buildPathRelativeToHudsonRoot(file), file.isDirectory());
+	}
+
+	public boolean matches(Saveable saveable, String pathRelativeToRoot, boolean isDirectory) {
+		if (pathRelativeToRoot != null) {
+			// Guard our own SCM workspace and the war directory. User-defined includes might inadvertently include those if they start with * or **!
+			if (pathRelativeToRoot.equals(SCM_WORKING_DIRECTORY) || pathRelativeToRoot.startsWith(SCM_WORKING_DIRECTORY + '/')) {
+				return false;
+			} else if (pathRelativeToRoot.equals(WAR_DIRECTORY) || pathRelativeToRoot.startsWith(WAR_DIRECTORY + '/')) {
+				return false;
+			}
+	        AntPathMatcher matcher = new AntPathMatcher();
+	        String directoryName = null;
+	        for (String pattern : includesPatterns) {
+	            if (matcher.match(pattern, pathRelativeToRoot)) {
+	                return true;
+	            } else if (isDirectory) {
+	            	// pathRelativeFromRoot is be a directory, and the pattern end in a file name. In this case, we must claim a match.
```

Comment:
I would like to see your PR merged then I can replay the scenario I mentioned and see whether it is still the case. It may be that the problem is not there anymore once your PR gets merged. The functionality is there and everything worked, there was just an odd log on one of the exhaustive pattern matchers, which, you may have removed now altogether.

I will go with your PR!